onadata/libs/tests/serializers/test_dataview_serializer.py

```diff
@@ -46,9 +46,41 @@ def get_has_hxl_support_value(data_view_name):
 
             return data.get('has_hxl_support')
 
-        self.assertFalse(get_has_hxl_support_value('name_only'))
-        self.assertTrue(get_has_hxl_support_value('age_only'))
-        self.assertTrue(get_has_hxl_support_value('age_and_name'))
+        self.assertFalse(get_has_hxl_support_value("name_only"))
+        self.assertTrue(get_has_hxl_support_value("age_only"))
+        self.assertTrue(get_has_hxl_support_value("age_and_name"))
+
+    def test_no_error_thrown_for_a_dataview_with_deleted_by_set(self):
+        """
+        When `deleted_by` is set, don't throw error when serializing a DataView
+        """
+        self._publish_form_with_hxl_support()
+        request = self.factory.get("/", **self.extra)
+        request.user = self.user
+
+        dataview_name = "My deleted DataView"
+        payload = {
+            "name": dataview_name,
+            "xform": "http://testserver/api/v1/forms/%s" % self.xform.pk,
+            "project": "http://testserver/api/v1/projects/%s" % self.project.pk,
+            "columns": '["name", "age"]',
+            "query": "[]",
+        }
+        serializer = DataViewSerializer(data=payload, context={"request": request})
+        is_valid = serializer.is_valid()
+        self.assertTrue(is_valid)
+
+        serializer.save()
+        self.assertEqual(DataView.objects.count(), 1)
+
+        # delete dataview and check dataviews endpoint
+        dataview = DataView.objects.get(name=dataview_name)
+        dataview.soft_delete(self.user)
+        serialized_dataview = DataViewSerializer(
+            dataview, context={"request": request}
+        ).data
+        self.assertFalse("deleted_by" in serialized_dataview)
+        self.assertFalse("deleted_at" in serialized_dataview)
```

Comment:
I think the `deleted_*` should not be present in the serializer. We may need to explicitly exclude the fields from the serializer. Note it will show up even for data views that have yet to be soft deleted.

Reply:
@ukanga Looks like the fields had been initially included in the `fields` serializer but have now been removed.

Reply:
@FrankApiyo @ukanga We do not need this test since we have filtered out deleted records and removed the `deleted_at`, `deleted_by` fields from the serializer